plugins/bundle/plugin.go

```diff
@@ -303,6 +303,35 @@ func (p *Plugin) activate(ctx context.Context, name string, b *bundle.Bundle) er
 			}
 		}
 
+		// Before changing anything make sure the roots don't collide with any
```

Comment:
Nit: can we move this logic into another function and call from here? I think it'll help with readability.

Reply:
Can do. I didn't originally because most of this gets refactored in the upcoming changes for #1584 but I guess thats just me being lazy 😄